src/server/pachyderm_test.go

```diff
@@ -9150,6 +9151,86 @@ func TestSpout(t *testing.T) {
 			}
 		}
 	})
+	t.Run("ServiceSpout", func(t *testing.T) {
+		dataRepo := tu.UniqueString("TestServiceSpout_data")
+		require.NoError(t, c.CreateRepo(dataRepo))
+
+		annotations := map[string]string{"foo": "bar"}
+
+		pipeline := tu.UniqueString("pipelineservicespout")
+		_, err := c.PpsAPIClient.CreatePipeline(
+			c.Ctx(),
+			&pps.CreatePipelineRequest{
+				Pipeline: client.NewPipeline(pipeline),
+				Transform: &pps.Transform{
+					Image: "pachyderm/ubuntuplusnetcat:latest",
+					Cmd:   []string{"sh"},
+					Stdin: []string{
+						"netcat -l -s 0.0.0.0 -p 8000 >/pfs/out",
+					},
+				},
+				ParallelismSpec: &pps.ParallelismSpec{
+					Constant: 1,
+				},
+				Input:  client.NewPFSInput(dataRepo, "/"),
+				Update: false,
+				Spout: &pps.Spout{
+					Service: &pps.Service{
+						InternalPort: 8000,
+						ExternalPort: 31800,
+						Type:         "NodePort",
```

Comment:
As asked in slack (https://pachyderm.slack.com/archives/CDWDPKPMY/p1560534040273200), I'd like the service within the spout field to default to type "NodePort" just as a regular pipeline service does, but I couldn't find where we actually do that, if any of you guys happen to know?

Reply:
it's here: https://github.com/pachyderm/pachyderm/blob/master/src/server/pps/server/api_server.go#L1549

Reply:
it's here: https://github.com/pachyderm/pachyderm/blob/master/src/server/pps/server/api_server.go#L2135

Reply:
Dope, thanks!